src/easyscience/base_classes/new_base.py

```diff
@@ -0,0 +1,145 @@
+from __future__ import annotations
+
+#  SPDX-FileCopyrightText: 2025 EasyScience contributors  <core@easyscience.software>
+#  SPDX-License-Identifier: BSD-3-Clause
+#  © 2021-2025 Contributors to the EasyScience project <https://github.com/easyScience/EasyScience
+from inspect import getfullargspec
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Any
+    from typing import Dict
+    from typing import Iterable
+    from typing import List
+    from typing import Optional
+    from typing import Set
+
+from easyscience import global_object
+
+from ..global_object.undo_redo import property_stack
+from ..io.serializer_base import SerializerBase
+
+
+class NewBase:
+    """
+    This is the new base class for easyscience objects.
+    It provides serialization capabilities as well as unique naming and display naming.
+    """
```

Comment:
You might consider using `__slots__` for memory efficiency. This tells the interpreter which attributes the instances are allowed to have. This also removes `__dict__`, which is normally used to store attributes dynamically.
Here, it would be 
```python
__slots__ =  ['_global_object', '_unique_name', '_default_unique_name', '_display_name'] 
```